ext/afform/core/Civi/Afform/Tokens.php

```diff
@@ -65,6 +65,9 @@ public static function hook_civicrm_tokens(GenericHookEvent $e) {
       $e->tokens['afform']["afform.{$tokenName}Url"] = E::ts('%1 (URL)', [1 => $afform['title'] ?? $afform['name']]);
       $e->tokens['afform']["afform.{$tokenName}Link"] = E::ts('%1 (Full Hyperlink)', [1 => $afform['title'] ?? $afform['name']]);
     }
+
+    $e->tokens['afform']["afform.validateSubmissionUrl"] = E::ts('Validate Submission URL');
+    $e->tokens['afform']["afform.validateSubmissionLink"] = E::ts('Validate Submission (Full Hyperlink)');
```

Comment:
This is causing a [weird test failure](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/4938/BKPROF=dfl,SUITES=phpunit-crm,label=bknix-tmp/testReport/junit/(root)/CRM_Core_BAO_MessageTemplateTest/testContactTokens/)... also I don't quite understand these tokens - they don't contain the name of the form... how would they be used?

Reply:
I have added these tokens but not the evaluation code hence the test are failing.

These tokens are for embedding the URL that will allow verification for the submission. I have not included the form name as verification is done via submission id.